src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -456,6 +457,22 @@ private void RenderTagOutput(string tagOutputMethodName)
                    .WriteEndMethodInvocation();
         }
 
+        private void RenderSelfClosing(bool selfClosing)
+        {
+            if (_designTimeMode)
+            {
+                return;
+            }
+            if (selfClosing == false)
```

Comment:
Lets not assume `selfClosing = false` for the `TagHelperExecutionContext` since it's replaceable. Always set the value.